druid-shell/src/backend/gtk/dialog.rs

```diff
@@ -98,9 +98,14 @@ pub(crate) fn get_file_dialog_path(
     let result = dialog.run();
 
     let result = match result {
-        ResponseType::Accept => match dialog.filename() {
-            Some(path) => Ok(path.into_os_string()),
-            None => Err(anyhow!("No path received for filename")),
+        ResponseType::Accept => match dialog.filenames() {
+            filenames if filenames.is_empty() => Err(anyhow!("No path received for filename")),
+            // This really shouldn't happen, unless there is a bug
+            // when setting the dialog options
+            filenames if filenames.len() > 1 && !options.multi_selection => {
```

Comment:
One last corner case that I noticed because of the `unwrap` below: if they pass `FileDialogType::Save` and `options.multi_selection = true` (which they shouldn't but I don't think we prevent it), then we correctly handle that combination on line 62 (by ignoring multi selection) but we don't check here that we only got one file back. Ok, maybe that's a bit paranoid, but I don't think it's any more paranoid than the existing checks that we have here...

Reply:
I'm not sure I understand what you mean:

> if they pass FileDialogType::Save and options.multi_selection = true, then we correctly handle that combination on line 62  but we don't check here that we only got one file back

We do check that the we only got one file back (`filenames.is_empty` and `filenames.len() > 1`), and return an `Err` if that is not true.

Do you mean that since we ignored the option in the case of a Save dialog, we should not return an error, but the whole `filenames` vec (essentially avoid this check)?

Reply:
What I meant was: in the case of a Save dialog, we should return an error if there's more than one filename, even if `options.multi_selection` is true. (This is what the `unwrap` at line `945` assumes)

Reply:
Ah, ok, so instead of checking for `&& !options.multi_selection` we should do:
```rust
            filenames if filenames.len() > 1 && action == FileChooserAction::Save => {
```

to uphold what we write later in the other comment, right?

Reply:
Yes, but you will need to check both the action *and* `options.multi_selection` because we still need to handle the case of open dialogs that expect a single file.

Reply:
Ok, fixing it now